pkg/cmd/ssh/ssh_test.go

```diff
@@ -634,31 +660,69 @@ var _ = Describe("SSH Command", func() {
 	})
 
 	Describe("ValidArgsFunction", func() {
+		var (
+			manager *targetmocks.MockManager
+			client  *gardenclientmocks.MockClient
+		)
+
 		BeforeEach(func() {
-			monitoringNode := &corev1.Node{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: "monitoring",
-				},
-			}
+			manager = targetmocks.NewMockManager(ctrl)
+			client = gardenclientmocks.NewMockClient(ctrl)
 
-			workerNode := &corev1.Node{
+			monitoringMachine := &machinev1alpha1.Machine{
 				ObjectMeta: metav1.ObjectMeta{
-					Name: "worker",
+					Name:      "monitoring1",
+					Namespace: "shoot--prod1--test-shoot",
+					Labels:    map[string]string{"node": "monitoring1"},
 				},
 			}
 
-			shootClient = internalfake.NewClientWithObjects(monitoringNode, workerNode)
+			seedClient = internalfake.NewClientWithObjects(testMachine, monitoringMachine)
+			shootClient = internalfake.NewClientWithObjects(testNode)
+
+			factory.ManagerImpl = manager
+			manager.EXPECT().CurrentTarget().Return(currentTarget, nil)
+			manager.EXPECT().GardenClient(currentTarget.GardenName()).Return(client, nil)
+
+			client.EXPECT().FindShoot(ctx, currentTarget.AsListOption()).Return(testShoot, nil)
+		})
+
+		It("should return all names based on machine objects", func() {
+			manager.EXPECT().SeedClient(ctx, gomock.Any()).Return(seedClient, nil)
+
+			options := ssh.NewSSHOptions(streams)
+			cmd := ssh.NewCmdSSH(factory, options)
+
+			suggestions, directive := cmd.ValidArgsFunction(cmd, nil, "")
+			Expect(directive).To(Equal(cobra.ShellCompDirectiveNoFileComp))
+			Expect(suggestions).To(HaveLen(2))
+			Expect(suggestions).To(Equal([]string{"node1", "monitoring1"}))
 		})
 
-		It("should find nodes based on their prefix", func() {
+		It("should return all names based on node objects", func() {
+			errForbidden := &apierrors.StatusError{ErrStatus: metav1.Status{Reason: metav1.StatusReasonForbidden}}
+			manager.EXPECT().SeedClient(gomock.Any(), gomock.Any()).Return(nil, errForbidden)
+			manager.EXPECT().ShootClient(gomock.Any(), gomock.Any()).Return(shootClient, nil)
```

Comment:
```suggestion
			manager.EXPECT().SeedClient(ctx, gomock.Any()).Return(nil, errForbidden)
			manager.EXPECT().ShootClient(ctx, currentTarget).Return(shootClient, nil)
```
Is this not working?

Reply:
right, I overlooked it - sorry
✅